app/models/accounting/quickbooks/transaction_creator.rb

```diff
@@ -78,6 +85,17 @@ def create_customer_reference(qb_customer_id:)
         entity
       end
 
+      def find_or_create_qb_customer_id_from(organization:)
+        return organization.qb_id if organization.qb_id.present?
+
+        qb_customer = ::Quickbooks::Model::Customer.new
+        qb_customer.display_name = organization.name
+
+        new_qb_customer = customer_service.create(qb_customer)
+
+        new_qb_customer.id
+      end
+
```

Comment:
And in general, please try to be careful about adding too much responsibility to a class like this. If you're creating and managing customers inside something called TransactionCreator, that is a hint that you need a new class.